### PR TITLE
Frontend/fix : Added functionality to toggle demo mode

### PR DESCRIPTION
**Pull request by aryan-13**
## Type of Change

- [ ] Bugfix
- [ ] New feature
- [x] Enhancement
- [ ] Refactoring
- [ ] Dependency updates

## Description

Added functionality to toggle demo mode

### Additional Changes

- [ ] This PR modifies the database schema (database migration added)
- [ ] This PR modifies dhall configs/environment variables
- [ ] This PR contains API breaking changes

## Checklist

- [ ] I formatted the code and addressed linter errors `./dev/format-all-files.sh`
- [ ] I reviewed submitted code
- [ ] I added unit tests for my changes where possible
- [ ] I added a [CHANGELOG](/CHANGELOG.md) entry if applicable

### CODE REVIEW

File: Frontend/ui-driver/src/Flow.purs [hunk 1]
```diff
@@ -63,6 +63,7 @@ baseAppFlow = do
     versionCode <- lift $ lift $ liftFlow $ getVersionCode
     checkVersion versionCode
     cacheAppParameters versionCode
+    _ <- pure $ setValueToLocalStore IS_DEMOMODE_ENABLED "false"
```
Comment (witcher-shailesh) [inline Frontend/ui-driver/src/Flow.purs:66]: why on every app load we are disabling demo mode?

File: Frontend/ui-driver/src/Screens/HomeScreen/Controller.purs [hunk 2]
```diff
@@ -261,7 +261,12 @@ eval (Notification notificationType) state = do
 eval CancelGoOffline state = do 
   continue state { props = state.props { goOfflineModal = false } }
 
-eval (GoOffline status) state = exit (DriverAvailabilityStatus state { props = state.props { goOfflineModal = false }} ST.Offline)
+eval (GoOffline status) state = do
+  when ((getValueToLocalStore IS_DEMOMODE_ENABLED) == "true") $ do
```
Comment (witcher-shailesh) [inline Frontend/ui-driver/src/Screens/HomeScreen/Controller.purs:265]: why are we doing with when conditions? It's a either case

File: Frontend/ui-driver/src/Screens/HomeScreen/Controller.purs [hunk 3]
```diff
@@ -477,7 +478,15 @@ eval (RideActiveAction activeRide) state = updateAndExit state { data {activeRid
 eval RecenterButtonAction state = continue state
 
 eval (SwitchDriverStatus status) state = do
-  if state.props.driverStatusSet == status then continue state 
+  if (((getValueToLocalStore IS_DEMOMODE_ENABLED) == "true") && (status == ST.Silent || status == ST.Offline)) then do
```
Comment (witcher-shailesh) [inline Frontend/ui-driver/src/Screens/HomeScreen/Controller.purs:481]: What if he is ONLINE?

File: Frontend/ui-driver/src/Screens/HomeScreen/Controller.purs [hunk 4]
```diff
@@ -487,7 +496,8 @@ eval (SwitchDriverStatus status) state = do
             let checkIfLastWasSilent = state.props.driverStatusSet == ST.Silent
             continue state { props { goOfflineModal = checkIfLastWasSilent, silentPopUpView = not checkIfLastWasSilent }}
 
-eval (PopUpModalSilentAction (PopUpModal.OnButton1Click)) state = exit (DriverAvailabilityStatus state{props{silentPopUpView = false}} ST.Offline)
+eval (PopUpModalSilentAction (PopUpModal.OnButton1Click)) state = do
```
Comment (witcher-shailesh) [inline Frontend/ui-driver/src/Screens/HomeScreen/Controller.purs:499]: why do is required here?

File: Frontend/ui-driver/src/Screens/HomeScreen/Controller.purs (hunk 3 above)
Comment (aryan-13) [reply Frontend/ui-driver/src/Screens/HomeScreen/Controller.purs:481]: Do we need to disable the demo mode on pressing the online tab as well?

**Review by witcher-shailesh — APPROVED** [review]

**Pull request merged**